.travis.yml

```diff
@@ -29,13 +32,68 @@ matrix:
       - python: pypy3
         env: TOX_ENV=pypy3
 
+# for instrumental we're checking if the coverage changed from base branch
+# so collect that info
+before_install:
+  - |
+       echo -e "TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST\n" \
+               "TRAVIS_REPO_SLUG=$TRAVIS_REPO_SLUG\n" \
+               "TRAVIS_PULL_REQUEST=$TRAVIS_PULL_REQUEST\n" \
+               "TRAVIS_COMMIT=$TRAVIS_COMMIT\n" \
+               "TRAVIS_PYTHON_VERSION=$TRAVIS_PYTHON_VERSION"
+  - |
+       # workaround https://github.com/travis-ci/travis-ci/issues/2666
+       if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+         URL="https://github.com/${TRAVIS_REPO_SLUG}/pull/${TRAVIS_PULL_REQUEST}.patch"
+         # `--location` makes curl follow redirects
+         PR_FIRST=$(curl --silent --show-error --location $URL | head -1 | grep -o -E '\b[0-9a-f]{40}\b' | tr -d '\n')
+         TRAVIS_COMMIT_RANGE=$PR_FIRST^..$TRAVIS_COMMIT
+       fi
+  # sanity check current commit
+  - git rev-parse HEAD
+  - echo "TRAVIS_COMMIT_RANGE=$TRAVIS_COMMIT_RANGE"
+  - git fetch origin master:refs/remotes/origin/master
+
+
 install:
   - pip list
-  - if [[ -e build-requirements-${TRAVIS_PYTHON_VERSION}.txt ]]; then travis_retry pip install -r build-requirements-${TRAVIS_PYTHON_VERSION}.txt; else travis_retry pip install -r build-requirements.txt; fi
+  - |
+      if [[ -e build-requirements-${TRAVIS_PYTHON_VERSION}.txt ]]; then
+        travis_retry pip install -r build-requirements-${TRAVIS_PYTHON_VERSION}.txt;
+      else
+        travis_retry pip install -r build-requirements.txt;
+      fi
+  - if [[ $INSTRUMENTAL ]]; then travis_retry pip install instrumental; fi
   - pip list
 script:
-  - tox -e $TOX_ENV
+  - if [[ $TOX_ENV ]]; then tox -e $TOX_ENV; fi
   - tox -e speed
+  - cp diff-instrumental.py diff-instrumental-2.py
+  - |
+      if [[ $INSTRUMENTAL && $TRAVIS_PULL_REQUEST != "false" ]]; then
+        git checkout $PR_FIRST^
+        # exclude the super slow test_malformed_sigs.py, until #127 is merged
+        files="src/ecdsa/test_der.py src/ecdsa/test_ecdsa.py src/ecdsa/test_ellipticcurve.py src/ecdsa/test_numbertheory.py src/ecdsa/test_pyecdsa.py"
```

Comment:
What if new test scripts are added? Wouldn't it be better to really exclude the slow test rather than "include everything else"?